### DISCUSSION TITLE
Develop on windows, install on Linux

### DISCUSSION THREAD

**Discussion by RachelAlcraft** [AvaloniaUI/Avalonia#5560] [Q&A] [answered]
Hi I have no difficulty developing the xaml and code for my app on a windows machine, but the tutorial doesn't take me from there to actually installing it on linux. I tried using "publish 1-click" but it didn't even work on windows. I have now tried getting docker to work, but not sure it is the way I am supposed to be going so not sure if I should be trying to figure out how it works or am going in the wrong direction - can anyone help? Thanks.

**Comment by kekekeks** [comment, marked as answer]
1) `dotnet publish -r linux-x64 -o path-to-output-directory`
2) copy to the target machine
3) `chmod +x YourProjectName`
4) `./YourProjectName`

For properly packaging your app you can tinker with https://github.com/qmfrederik/dotnet-packaging but that requires some knowledge about deb and rpm packages.

**Reply by RachelAlcraft** [reply]
It works, that's amazing, thanks!
